control/controlclient/direct.go

```diff
@@ -176,6 +176,21 @@ func (c *Direct) SetHostinfo(hi *tailcfg.Hostinfo) {
 	c.hostinfo = hi.Clone()
 }
 
+// SetNetInfo clones the provided NetInfo and remembers it for the
+// next update.
+func (c *Direct) SetNetInfo(ni *tailcfg.NetInfo) {
+	if ni == nil {
+		panic("nil NetInfo")
+	}
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.logf("NetInfo: %v\n", ni)
+	if c.hostinfo != nil {
```

Comment:
Should we silently drop NetInfo here, or create a c.hostinfo?

Reply:
I added a log that it's being ignored instead. It really shouldn't be called with hostinfo nil.

Filed https://github.com/tailscale/tailscale/issues/136 instead to add a log convention for such things.